test/foundry/abstract/ERC20PriceOracleReceiptVaultTest.sol

```diff
@@ -0,0 +1,88 @@
+// SPDX-License-Identifier: CAL
+pragma solidity =0.8.25;
+
+import {Test, Vm} from "forge-std/Test.sol";
+import {ICloneableFactoryV2} from "rain.factory/interface/ICloneableFactoryV2.sol";
+import {CloneFactory} from "rain.factory/concrete/CloneFactory.sol";
+import {
+    ERC20PriceOracleReceiptVault,
+    ReceiptVaultConstructionConfig
+} from "contracts/concrete/vault/ERC20PriceOracleReceiptVault.sol";
+import {LibERC20PriceOracleReceiptVaultCreator} from "../lib/LibERC20PriceOracleReceiptVaultCreator.sol";
+import {Receipt as ReceiptContract} from "contracts/concrete/receipt/Receipt.sol";
+import {TestErc20} from "contracts/test/TestErc20.sol";
+import {TwoPriceOracle, TwoPriceOracleConfig} from "contracts/oracle/price/TwoPriceOracle.sol";
+import {
+    ChainlinkFeedPriceOracle,
+    ChainlinkFeedPriceOracleConfig
+} from "contracts/oracle/price/chainlink/ChainlinkFeedPriceOracle.sol";
+import {MockChainlinkDataFeed, RoundData} from "contracts/test/MockChainlinkDataFeed.sol";
+
+contract ERC20PriceOracleReceiptVaultTest is Test {
+    ICloneableFactoryV2 internal immutable iFactory;
+    ERC20PriceOracleReceiptVault internal immutable iImplementation;
+    ReceiptContract internal immutable receiptImplementation;
```

Comment:
if this is immutable we should use the `iFoo` naming convention e.g. `iReceiptImplementation`